### PR TITLE
Added tests for compute_position_assignments()

### PR DESCRIPTION
**Pull request by dkgoutham**
- Added comprehensive tests for compute_position_assignments() including edge cases
- Refactored the defrida_proofs tests a bit.
- ~TODO: Fix tests: test calculation::tests::test_get_min_max_mean_from_all_benchmark_metrics ... FAILED, test handlers::tests::test_handler ... FAILED~ Done
- ~TODO: Before merging, need to fix issues reported by linter and "cargo deny check"!~ Can relax this!

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ resolver = "3"
 
 [workspace.dependencies]
 hotstuff_rs = "0.4.0"
-frida-poc = { git = "https://github.com/NethermindEth/Frida-poc.git", branch = "main" }
+frida-poc = { git = "ssh://git@github.com/NethermindEth/Frida-poc.git" }
```
Comment (maksimryndin) [inline Cargo.toml]: why ssh? why not git with a specific commit?

File: crates/benchmark/src/calculation.rs [hunk 2]
```diff
@@ -571,49 +571,49 @@ mod tests {
             PhaseTimingAndProofSize::get_min_max_mean_from_all_benchmark_metrics(all_metrics);
 
         {
-            assert_eq!(metrics.propose_block_time.min_time.unwrap(), 0);
-            assert_eq!(metrics.propose_block_time.mean_time.unwrap(), 1);
+            assert_eq!(metrics.propose_block_time.min_time.unwrap(), 3);
```
Comment (maksimryndin) [inline crates/benchmark/src/calculation.rs:576]: how the test is fixed? How should we justify these values? (I want to be sure that we don't just fit under the asserts)

File: Cargo.toml (hunk 1 above)
Comment (dkgoutham) [reply Cargo.toml]: We can, I thought ssh was a better approach! It will always pull from the main branch.

Comment (maksimryndin) [reply Cargo.toml]: No, git resolution is better as it allows to pin versions

Comment (dkgoutham) [reply Cargo.toml]: Sure, will update it once frida-poc is finalised and merged.

**Review by maksimryndin — APPROVED** [review]
